tests/integration/test_definitions.py

```diff
@@ -45,18 +54,80 @@ def define_tests():
-            "merlin server start; merlin server status; merlin server stop",
+            """merlin server init;
+            merlin server start;
+            merlin server status;
+            merlin server stop;""",
+            [
+                HasRegex("Server started with PID [0-9]*"),
+                HasRegex("Merlin server is running"),
+                HasRegex("Merlin server terminated"),
+            ],
+            "local",
+            "rm -rf appendonly.aof dump.rdb merlin_server/",
+        ),
+        "merlin server restart": (
+            """merlin server init;
+            merlin server start;
+            merlin server restart;
+            merlin server status;
+            merlin server stop;""",
             [
                 HasRegex("Server started with PID [0-9]*"),
                 HasRegex("Merlin server is running"),
                 HasRegex("Merlin server terminated"),
             ],
             "local",
+            "rm -rf appendonly.aof dump.rdb merlin_server/",
+        ),
+    }
+    server_config_tests = {
+        "merlin server change config": (
+            """merlin server init;
+            merlin server config -p 8888 -pwd new_password -d ./config_dir -ss 80 -sc 8 -sf new_sf -am always -af new_af.aof;
+            merlin server start;
+            merlin server stop;""",
+            [
+                FileHasRegex("merlin_server/redis.conf", "port 8888"),
+                FileHasRegex("merlin_server/redis.conf", "requirepass new_password"),
+                FileHasRegex("merlin_server/redis.conf", "dir ./config_dir"),
+                FileHasRegex("merlin_server/redis.conf", "save 80 8"),
+                FileHasRegex("merlin_server/redis.conf", "dbfilename new_sf"),
+                FileHasRegex("merlin_server/redis.conf", "appendfsync always"),
+                FileHasRegex("merlin_server/redis.conf", 'appendfilename "new_af.aof"'),
+                PathExists("./config_dir/new_sf"),
+                PathExists("./config_dir/appendonlydir"),
+                HasRegex("Server started with PID [0-9]*"),
+                HasRegex("Merlin server terminated"),
+            ],
+            "local",
+            "rm -rf appendonly.aof dump.rdb merlin_server/ config_dir/",
+        ),
+        "merlin server config add/remove user": (
+            """merlin server init;
+            merlin server start;
+            merlin server config --add-user new_user new_password;
+            merlin server stop;
+            scp ./merlin_server/redis.users ./merlin_server/redis.users_new
+            merlin server start;
+            merlin server config --remove-user new_user;
+            merlin server stop;
+            """,
+            [
+                FileHasRegex("./merlin_server/redis.users_new", "new_user"),
+                FileHasNoRegex("./merlin_server/redis.users", "new_user"),
+            ],
+            "local",
+            """rm -rf appendonly.aof dump.rdb merlin_server/""",
```

Comment:
This command is called a lot. Is it worth storing this as a variable?

Reply:
The command is used for cleaning the files directories created by merlin server and this ensures that new merlin server tests starts new without any artifacts. There might be some changes to this command if configurations are changed, but I agree with you that it might be worthwhile to have a variable for this command. I have added one as a constant at the top.